theme/snippets/image-caption-module.liquid

```diff
@@ -22,17 +27,36 @@
       col-12 sm:col-9
       {% endif %}
     " 
-    src="{{ image_module_image.src }}" 
-    alt="{{ image_module_image.alt }}
+    src="{{ image_module_first_image.src }}" 
+    alt="{{ image_module_first_image.alt }}
     "/>
     {% if caption %}
-      <span class="{{ color }} pt_25 sans-xs
+      <span class="{{ color }} pt_25 sm:pt0 sans-xs
         {% if variant == 'left' %}
-        col-12 sm:col-3 sm:pl2_5
+        col-12 sm:col-3 sm:ml2_5
         {% endif %}
         ">
         {{ caption }}
       </span>
     {% endif %}
   </div>
 {% endif %}
+
+
+{% if variant == 'two-up' and image_module_first_image and image_module_last_image %}
+  <div class="ImageCaptionModule flex flex-col xs:flex-row py3">
+    <div class="ImageCaptionModule__images--style-two-up flex flex-row xs:col-9">
+      {% for image in image_module_images %}
+        <div class="col-6">
+          <img class="w100 h100 fit-cover" src="{{ image.src }}" alt="{{ image.alt }}"/>
+        </div>
+      {% endfor %}
+    </div>
+    {% if caption %}
+      <span class="flex {{ color }} pt_25 xs:pt0 sans-xs
+        col-10 xs:col-3 xs:ml2_5">
+        {{ caption }}
+      </span>
+    {% endif %}
+  </div>
+{% endif %}
```

Comment:
I probably could have made this work adding it to the code above, with lots of if statements, but I felt like it was getting hard to read/understand. thoughts?